manifests/server/passwd.pp

```diff
@@ -5,6 +5,15 @@
   $group             = $postgresql::server::group
   $psql_path         = $postgresql::server::psql_path
   $port              = $postgresql::server::port
+  $database          = $postgresql::server::default_database
+
+  # psql will default to connecting as $user if you don't specify name
+  # dbname option could probably be used unconditionally assuming it is in all supported versions of psql
+  $datbase_user_same = $database == $user
+  $dboption = $datbase_user_same ? {
```

Comment:
* You can shorten that by putting `($database == $user)` directly into the selector.
* Using `$_dboption` will make it a private variable, and avoid unintentional linkage later.